test/ReadsSet_basic_test.py

```diff
@@ -1,177 +1,154 @@
+        "description": set_description,
+        "items": set_items,
+    }
 
-    assert "ref_path" in item2
-    assert item2["ref_path"] == res["set_ref"] + ";" + item2["ref"]
+    result = save_reads_set(set_api_client, context, ws_id, set_name, set_data)
 
+    return {
+        "obj": result,
+        "set_name": set_name,
+        "set_type": API_CLASS.set_type(),
+        "set_description": set_description,
+        "n_items": len(reads_refs),
+        "second_set_item": set_items[1],
+    }
 
-def test_save_and_get_of_empty_set(
-    set_api_client: SetAPI, context: dict[str, str | list], ws_id: int
-) -> None:
-    set_name = "nada_set"
-    set_description = "nothing to see here"
-    n_items_in_set = 0
 
-    # create the set object
-    set_data = {"description": set_description, "items": []}
+@pytest.fixture(scope="module")
+def empty_reads_set(
+    set_api_client: SetAPI,
+    context: dict[str, str | list],
+    ws_id: int,
+) -> dict[str, Any]:
+    """A set with no description and an empty items list."""
+    set_name = "empty_reads_set"
+    # omit the set description and make the set items an empty list
+    set_data = {
+        "items": [],
+    }
 
-    # test a save
-    res = set_api_client.save_reads_set_v1(
-        context,
-        {
-            "data": set_data,
-            "output_object_name": set_name,
-            "workspace_id": ws_id,
-        },
-    )[0]
-    assert "set_ref" in res
-    assert "set_info" in res
-    assert len(res["set_info"]) == INFO_LENGTH
-
-    assert res["set_info"][1] == set_name
-    assert "item_count" in res["set_info"][10]
-    assert res["set_info"][10]["item_count"] == str(n_items_in_set)
-
-    # test get of that object
-    d1 = set_api_client.get_reads_set_v1(context, {"ref": res["set_ref"]})[0]
-    assert "data" in d1
-    assert "info" in d1
-    assert len(d1["info"]) == INFO_LENGTH
-    assert "item_count" in d1["info"][10]
-    assert d1["info"][10]["item_count"] == str(n_items_in_set)
-
-    assert d1["data"]["description"] == set_description
-    assert len(d1["data"]["items"]) == 0
-
-    d2 = set_api_client.get_reads_set_v1(
-        context, {"ref": res["set_ref"], INC_ITEM_INFO: 1}
-    )[0]
+    result = save_reads_set(set_api_client, context, ws_id, set_name, set_data)
+
+    return {
+        "obj": result,
+        "set_name": set_name,
+        "set_type": API_CLASS.set_type(),
+        # the class fills in the missing description field
+        "set_description": "",
+        "n_items": 0,
+    }
 
-    assert "data" in d2
-    assert "info" in d2
-    assert len(d2["info"]) == INFO_LENGTH
-    assert "item_count" in d2["info"][10]
-    assert d2["info"][10]["item_count"] == str(n_items_in_set)
 
-    assert d2["data"]["description"] == set_description
-    assert len(d2["data"]["items"]) == n_items_in_set
+def test_save_reads_set(
+    reads_set: dict[str, Any],
+) -> None:
+    check_save_set_output(**reads_set)
 
 
-def test_get_sampleset_as_readsset(
```

Comment:
I deleted this test because you can get any set type from any of the `get_***_set_v1` functions -- they basically just function as "get workspace object" with a few bells and whistles if the object has a `["data"]["items"]` field